pkg/quota/admission/clusterresourceoverride/admission.go

```diff
@@ -0,0 +1,161 @@
+	a.ProjectCache = projectCache
+}
+
+func ReadConfig(configFile io.Reader) (*api.ClusterResourceOverrideConfig, error) {
+	if configFile == nil || reflect.ValueOf(configFile).IsNil() /* pointer to nil */ {
+		glog.V(5).Infof("ClusterResourceOverride has no config to read.")
+		return nil, nil
+	}
+	glog.V(5).Infof("ClusterResourceOverride about to read config:\n%v", configFile)
+	buffer := new(bytes.Buffer)
+	if _, err := buffer.ReadFrom(configFile); err != nil {
+		return nil, err
+	}
+	config := &api.ClusterResourceOverrideConfig{}
+	err := yaml.Unmarshal(buffer.Bytes(), config)
+	glog.V(5).Infof("ClusterResourceOverride config:\n%v\nerror: %v", config, err)
+	return config, err
+}
+
+func (a *clusterResourceOverridePlugin) Validate() error {
+	if err := validate.Validate(a.Config); err != nil {
+		return err
+	}
+	if a.ProjectCache == nil {
+		return fmt.Errorf("ClusterResourceOverride did not get a project cache")
+	}
+	return nil
+}
+
+// TODO this will need to update when we have pod requests/limits
+func (a *clusterResourceOverridePlugin) Admit(attr admission.Attributes) error {
+	glog.V(8).Infof("ClusterResourceOverride admission controller is invoked")
+	if a.Config == nil || attr.GetResource() != kapi.Resource("pods") || attr.GetSubresource() != "" {
+		return nil // not applicable
+	}
+	pod, ok := attr.GetObject().(*kapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attr, fmt.Errorf("unexpected object: %#v", attr.GetObject()))
+	}
+	glog.V(5).Infof("ClusterResourceOverride is looking at creating pod %s in project %s", pod.Name, attr.GetNamespace())
+
+	// allow annotations on project to override
+	if ns, err := a.ProjectCache.GetNamespace(attr.GetNamespace()); err != nil {
+		glog.Warningf("ClusterResourceOverride got an error retrieving namespace: %v", err)
+		return admission.NewForbidden(attr, err) // this should not happen though
+	} else {
+		projectEnabledPlugin, exists := ns.Annotations[clusterResourceOverrideAnnotation]
+		if exists && projectEnabledPlugin != "true" {
+			glog.V(5).Infof("ClusterResourceOverride is disabled for project %s", attr.GetNamespace())
+			return nil // disabled for this project, do nothing
+		}
+	}
+
+	// Reuse LimitRanger logic to apply limit/req defaults from the project. Ignore validation
+	// errors, assume that LimitRanger will run after this plugin to validate.
+	glog.V(5).Infof("ClusterResourceOverride: initial pod limits are: %#v", pod.Spec.Containers[0].Resources)
+	if err := a.LimitRanger.Admit(attr); err != nil {
+		glog.V(5).Infof("ClusterResourceOverride: error from LimitRanger: %#v", err)
+	}
+	pod, ok = attr.GetObject().(*kapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attr, fmt.Errorf("unexpected object: %#v", attr.GetObject()))
+	}
+	glog.V(5).Infof("ClusterResourceOverride: pod limits after LimitRanger are: %#v", pod.Spec.Containers[0].Resources)
+	for _, container := range pod.Spec.Containers {
+		resources := container.Resources
+		memLimit := resources.Limits.Memory()
```

Comment:
you should use form:

```
memLimit, found := resources.Limits[api.ResourceMemory]
```

This would let you distinguish if the value was set or if you are just getting the nil 0 value.